experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -487,6 +487,63 @@ export interface Sequence<TTypes extends AllowedTypes> extends TreeField {
 	 */
 	mapBoxed<U>(callbackfn: (value: TypedNodeUnion<TTypes>, index: number, array: this) => U): U[];
 
+	/**
+	 * Calls the specified callback function for all the elements in the sequence.
+	 * The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.
+	 * @param callbackfn - A function that accepts up to four arguments.
+	 * The reduce method calls the callbackfn function one time for each element in the array.
+	 * @param initialValue - If present, it is passed to the first call to the callbackfn.
+	 * Otherwise, the first element in the sequence will be used as the initial accumulator value and skipped as currentValue.
+	 */
+	reduce(
+		callbackfn: (
+			previousValue: UnboxNodeUnion<TTypes>,
+			currentValue: UnboxNodeUnion<TTypes>,
+			currentIndex: number,
+			sequence: this,
+		) => UnboxNodeUnion<TTypes>,
```

Comment:
Return type from callback and from the whole reduce function should be `U` not `UnboxNodeUnion<TTypes>` right?

Idea: define a simple class
```
export class Stop<T>{
constructor(public readonly value: T) {}
}
```

then make the return type from `callbackfn` `U | Stop<U>`, then th eimplementation can use `instanceof `Stop` to check for it and early exit (similar to the Skip pattern in the navigation file, but with a stored value to exit with).

That would allow for reduce to be used to implement "find" and "some" including the early exit optimization, and doesn't prevent any other use of reduce from working normally.
```

Reply:
Yes, you're right about the return type - fixed. There are two overloads of reduce, one which accumulates the same type as is in the array, and one which accumulates some other type "U". I had typed the latter overload wrongly for both boxed and unboxed.

And Stop class sounds good to me, I added it.